server/src/pages/shot/editor.js

```diff
@@ -262,27 +262,53 @@ exports.Editor = class Editor extends React.Component {
   }
 
   resizeCropBox(event, direction) {
+    let width = selectedPos.width;
+    let height = selectedPos.height;
     let rect = this.cropContainer.getBoundingClientRect();
     let diffX = event.clientX - rect.left - resizeStartPos.x;
     let diffY = event.clientY - rect.top - resizeStartPos.y;
     let movement = movementPositions[resizeDirection];
-    if (movement[0]) {
-        let moveX = movement[0];
-        moveX = moveX == "*" ? ["x1", "x2"] : [moveX];
-        for (let moveDir of moveX) {
-          selectedPos[moveDir] = this.truncateX(resizeStartSelected[moveDir] + diffX);
-        }
+    let checkLeftBorder = selectedPos.left == 0 && resizeStartSelected.left + diffX <= 0;
+    let checkRightBorder = selectedPos.right == this.canvasWidth && resizeStartSelected.right + diffX >= this.canvasWidth;
+    let checkTopBorder = selectedPos.top == 0 && resizeStartSelected.top + diffY <= 0;
+    let checkBottomBorder = selectedPos.bottom == this.canvasHeight && resizeStartSelected.bottom + diffY >= this.canvasHeight;
+    let checkResizeDirection = resizeDirection == "move";
```

Comment:
Nit: perhaps name these variables in a `isCondition` convention.  That'd read more naturally.